pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -402,9 +430,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	// completing. It can take an arbitrarily long time for a pod restart to successfully chown the
 	// directory. This is a race condition for all OSDs; therefore, do this in an init container.
 	// See more discussion here: https://github.com/rook/rook/pull/3594#discussion_r312279176
+	dataPath := ""
+	if doChownDataPath {
+		dataPath = osd.DataPath
+	}
 	initContainers = append(initContainers,
 		opspec.ChownCephDataDirsInitContainer(
-			opconfig.DataPathMap{ContainerDataDir: osd.DataPath},
+			opconfig.DataPathMap{ContainerDataDir: dataPath},
```

Comment:
Should this init container not be added if `dataPath == ""`? Otherwise, what is the behavior of this init container if the `ContainerDataDir` is empty?

Reply:
It still needs to chown the logging/crash dirs. If the data dir is empty, it merely omits chowning that dir.